src/parsing/compileAstToExpression.ts

```diff
@@ -1196,6 +1200,58 @@ function transformExpression(ast, compilationOptions) {
 	return new TransformExpression(transformCopies, modifyExpr, returnExpr);
 }
 
+function typeswitchExpr(ast: IAST, compilationOptions: CompilationOptions) {
+	const argExpr = compile(
```

Comment:
Could you add something similar as line 1097? an error that is thrown when we are not in 'XQuery' mode?

Reply:
Sure, but as I've said in a comment above, it is not working on that test (and in general, it's never executed).

Reply:
The coveralls tests are failing because of this, as the condition is never satisfied. We can just revert the merge if you consider it appropriate.